02.calendar/cal.rb

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env ruby
+require 'optparse'
+require 'date'
+
+# 表示カレンダーの横幅
+CALENDAR_WIDTH = 20
+
+# コマンドライン引数の取得
+begin
+  opt_params = ARGV.getopts('y:', 'm:')
+rescue OptionParser::MissingArgument => e
+  puts e.message
+  return
+end
+opt_y = opt_params["y"]
+opt_m = opt_params["m"]
+
+# オプション引数の範囲チェック
+if opt_y != nil && (1..9999).include?(opt_y.to_i) == false
+  puts "year `#{opt_y}' not in range 1..9999"
+  return
+end
+if opt_m != nil && (1..12).include?(opt_m.to_i) == false
+  puts  "#{opt_m} is neither a month number (1..12) nor a name"
+  return
+end
+
+# 表示するカレンダーの年月日と曜日
+today = Date.today.day
+disp_month = (opt_m != nil) ? opt_m.to_i : Date.today.month
+disp_year = (opt_y != nil) ? opt_y.to_i : Date.today.year
+disp_date = Date.new(disp_year, disp_month, 1)
+
+# 表示するカレンダーの最終日（=>翌月1日の前日）
+last_day = disp_date.next_month.prev_day.day
+
+# Key:日付, Value:日付に対して表示する文字列
+is_this_month = (disp_year == Date.today.year && disp_month == Date.today.month) ? true : false # 表示するのが今月のカレンダーか否か
```

Comment:
```suggestion
is_this_month = disp_year == Date.today.year && disp_month == Date.today.month
```

と等価かなと思います `==` や `&&` はそもそも `true` か `false` を返します。